byteutils/byteutils.go

```diff
@@ -15,3 +15,18 @@ func ReadAvailableBytesToBuffer(target []byte, targetOffset int, source []byte,
 
 	return bytesToRead
 }
+
+// Concat concatenates the byte slices into a new byte slice.
+func Concat(byteSlices ...[]byte) (result []byte) {
+	// sanitize parameters
+	if len(byteSlices) == 0 {
```

Comment:
Why is this a panic? `Concat()` returning `[]byte{}` seems perfectly fine to me.